### ISSUE TITLE
Allow people to set their own feed codes

### ISSUE THREAD

**Issue by nathansherburn** [SLC3/MARS#79]
Lecturers are finding it hard to remember the feed codes when students ask.

We'll need to restrict codes to be < 6 chars etc. so that students don't have to type sentences and guess characters to access a feed.

**Comment by aj1331** [comment]
The feed codes have to be unique as well. I don't think it's a good idea to
let them enter their own. The easy to remember ones will get taken quickly,
and as people have to pick more obscure things to find a unique code
they'll be harder to remember anyway with the added frustration of "The
code you have entered has already been used".

**Comment by nathansherburn** [comment]
Yeah I agree. One way to get around the "The code you have entered has already been used" problem is a generator. Similar to github's repository name maker but making shorter codes. As you said though, the short codes will go fast and you'll end up with stuff like "ENG1002_2016nathanS1" which will be a nightmare to type or a tonne of "already in use" messages.

Other solutions could be a feed search for students. Or simply just making the poll codes available for students to see on their phones (if they're not already?) and share.

**Comment by thanatos6666** [comment]
Firstly, let's make generated codes a bit more friendly to read - e.g. get rid of ambiguous 0 and O for instance.

Secondly, let's experiment with giving people the option of **9 character** codes. This allows for a unit name and short 2 character differentiator (ENG1002A, ENG1002B). 

If they're "already in use", tough luck, find something else. Let's see how we go.

**Comment by aj1331** [comment]
We need to consider how to handle cloned collections. Every collection's code has to be unique, so the new collection has to have a new code. If we're still going with the workflow of cloning and archiving between semesters that means the 9 character code can only be used for one semester, then the lecturer will have to come up with a new one.

**Comment by nathansherburn** [comment]
I suppose we could make it so that people can change the codes after the collection has been set.

For example, you want to archive the collection that currently has ENG1002A. You hit "Archive and Clone" but then realise that the ENG1002A code is used by the collection you just archived. So you go to the archived collection, change the feed code to 1002old1 and then use ENG1002A for the new collection.

We'd have to make sure we remove the audience from the feed everytime someone changes the collection name in this case. We'd need a warning to let the presenter know that this is going to happen too.

**Comment by nathansherburn** [comment]
I'm also looking at how we can make the feed code more prominent in the collection views. Veronica (our education designer in engineering) suggested that the quick start documentation could mention placing the feed code on the slides too.

I think combined this would ultimately solve the problem of people needing to set their own feed codes anyway but we'll have to see how it goes.

PS I also thought it could be cool to have a button to download a poll question as a slide (complete with mars formatting and feedcode etc). Discussion for another time though :p
